### DISCUSSION TITLE
install_github issue

### DISCUSSION THREAD

**Discussion by janekim319** [GreenleafLab/ArchR#895] [Questions / Documentation]
I have followed the instructions to download install ArchR on macOS. I have Big Sur and I'm having trouble even running install_github. I have devtools and biocmanager loaded in R. Below is the error message I get: 

Using github PAT from envvar GITHUB_PAT
Downloading GitHub repo GreenleafLab/ArchR@master
Error in utils::download.file(url, path, method = method, quiet = quiet,  : 
  download from 'https://api.github.com/repos/GreenleafLab/ArchR/tarball/master' failed

Does this error look familiar?

**Comment by rcorces** [comment]
This looks like it is coming from `remotes::install_github()` rather than `devtools::install_github()`.
Use
`devtools::install_github("GreenleafLab/ArchR", ref="master", repos = BiocManager::repositories())`

**Reply by janekim319** [reply]
Thank you for the reply! I actually have been using devtools::install_github. 

> devtools::install_github
function (repo, ref = "HEAD", subdir = NULL, auth_token = github_pat(quiet), 
    host = "api.github.com", dependencies = NA, upgrade = c("default", 
        "ask", "always", "never"), force = FALSE, quiet = FALSE, 
    build = TRUE, build_opts = c("--no-resave-data", "--no-manual", 
        "--no-build-vignettes"), build_manual = FALSE, build_vignettes = FALSE, 
    repos = getOption("repos"), type = getOption("pkgType"), 
    ...) 
pkgbuild::with_build_tools({
    ellipsis::check_dots_used(action = getOption("devtools.ellipsis_action", 
        rlang::warn))
    {
        remotes <- lapply(repo, github_remote, ref = ref, subdir = subdir, 
            auth_token = auth_token, host = host)
        install_remotes(remotes, auth_token = auth_token, host = host, 
            dependencies = dependencies, upgrade = upgrade, force = force, 
            quiet = quiet, build = build, build_opts = build_opts, 
            build_manual = build_manual, build_vignettes = build_vignettes, 
            repos = repos, type = type, ...)
    }
}, required = FALSE)
<bytecode: 0x7fe21d867120>
<environment: namespace:remotes>

Is it odd that the namespace shows up as remotes when I check the devtools::install_github() function?

**Reply by janekim319** [reply]
Here are more errors:

> devtools::install_github("GreenleafLab/ArchR", ref="master", repos = BiocManager::repositories())
Using github PAT from envvar GITHUB_PAT
Downloading GitHub repo GreenleafLab/ArchR@master
Error: Failed to install 'ArchR' from GitHub:
  Could not find tools necessary to compile a package
Call `pkgbuild::check_build_tools(debug = TRUE)` to diagnose the problem.
> pkgbuild::check_build_tools(debug = TRUE)
Trying to compile a simple C file
Error: Could not find tools necessary to compile a package
Call `pkgbuild::check_build_tools(debug = TRUE)` to diagnose the problem.
rror: Could not find tools necessary to compile a package

**Reply by rcorces** [reply]
this is not really my expertise. But nobody has encountered this error before so it must be something odd on your end.
Are you able to install anything from devtools?
for ex.
`devtools::install_github('immunogenomics/presto')`

**Reply by janekim319** [reply]
It might be a M1 chip Big Sur MacBook issue. I'm not sure. I'm able to use download other things using devtools::install_github. Do you think it could be that the most recent version of gfortran is for intel processors and it is not compatible with M1?

**Reply by rcorces** [reply]
Now that you mention that, maybe this helps
https://github.com/GreenleafLab/ArchR/issues/633

**Reply by janekim319** [reply]
Thank you for the suggestion! I followed everything on the link you provided with a small change from the stack overflow post. I used x86_64-apple-darwin20/11.1.0 instead of x86_64-apple-darwin19/10.2.0 in the Makevars file. 

I also ran it after running options(buildtools.check = function(action) TRUE ) to suppress the error "Could not find tools necessary to compile a package". Now I get many errors, mostly dependencies errors, but one stands out. Does "Rhdf5lib" sound familiar to you?

Error: package or namespace load failed for ‘Rhdf5lib’ in dyn.load(file, DLLpath = DLLpath, ...):
 unable to load shared object '/Library/Frameworks/R.framework/Versions/4.1-arm64/Resources/library/00LOCK-Rhdf5lib/00new/Rhdf5lib/libs/Rhdf5lib.so':
  dlopen(/Library/Frameworks/R.framework/Versions/4.1-arm64/Resources/library/00LOCK-Rhdf5lib/00new/Rhdf5lib/libs/Rhdf5lib.so, 6): no suitable image found.  Did find:
	/Library/Frameworks/R.framework/Versions/4.1-arm64/Resources/library/00LOCK-Rhdf5lib/00new/Rhdf5lib/libs/Rhdf5lib.so: mach-o, but wrong architecture
	/Library/Frameworks/R.framework/Versions/4.1-arm64/Resources/library/00LOCK-Rhdf5lib/00new/Rhdf5lib/libs/Rhdf5lib.so: mach-o, but wrong architecture
Error: loading failed
Execution halted

**Reply by rcorces** [reply]
I think you'll need to google around a bit to find your issues since these are not issues with ArchR and my only means of helping you is to google the same things that you could google.

**Reply by janekim319** [reply]
Thank you so much for your reply. My issue was that I was using the wrong version of R. I had to download R-4.1.0.pkg and not R-4.1.0-arm64.pkg! I hope this help anyone else going through the same issue with a macbook big sur M1 chip!

**Reply by mzhibo** [reply]
I also have this issue:
"Downloading GitHub repo GreenleafLab/ArchR@master
Error in utils::download.file(url, path, method = method, quiet = quiet,  : 
  download from 'https://api.github.com/repos/GreenleafLab/ArchR/tarball/master' failed"
I have been using ArchR for a while, previously it was fine. Not sure exactly when this error started. I running R in Linux. not on a macbook

**Reply by mzhibo** [reply]
This seems to be related to secure package download:
https://support.rstudio.com/hc/en-us/articles/206827897-Secure-Package-Downloads-for-R

`options(download.file.method = "wget")` worked for me (linux)
